### ISSUE TITLE
Release `v0.7`

### ISSUE THREAD

**Issue by jank324** [desy-ml/cheetah#165]
# Checklist for `v0.7` release features

 - [x] Vectorisation (#116)
 - [x] Add space charge (#142)
 - [ ] ~~Add nice interface for space charge~~ (delayed to later release) (#160)
 - [x] Add custom transfer map element (#100)
 - [x] Fix issue with default length (#116, #143, #164)
 - [x] Add uniformly distributed beam initialisation (#146)
 - [x] Fix NumPy 2 compatibility (#177)

Let's discuss what features we want to have ready before releasing `v0.7`.

**Comment by jank324** [comment]
One thought, now that we are a couple more people contributing to Cheetah, we should use this release to establish common ground on certain foundational design choices, e.g. the coordinate system. So if there is something like that you thing needs to be discussed, please bring it up here.

**Comment by cr-xu** [comment]
From the discussion before, the current plan is to  

- Fix remaining issues, review the code and merge #142
- Adapt the changes and make sure SC implementation is still correct in #163, then merge it into master
- Implement #160

**Comment by jank324** [comment]
I have some proposals on revising the plan for `v0.7.0`, considering that the version of Cheetah on `master` is by now very different from the most recent stable release you would get from PyPI ( `v0.6.3` without vectorisation and such).

 - #142 is merged.
 - #163 should definitely be merged before, being a bit of a breaking change and almost done.
 - I would suggest to defer #160 to `v0.7.1`
 - I think #208 I think is an important item we are working on right now, so it would be nice to have this in `v0.7.0`. But, considering that it will probably take some time, it might also make sense to make that part of `v0.7.1`.
 - #153 is almost ready, but it really doesn't matter if this appears in `v0.7.0` or `v0.7.1`.
 - I'm not sure where we stand with #164.
 - #213 is a bug fix for the vectorisation, so it should be merged before `v0.7.0` ... but it's ready, so that should be easy.

So I guess we could release `v0.7.0` as soon as #213 is merged and #163 is finalised and merged. The only open question is if we would want to wait for #208.

**Comment by cr-xu** [comment]
I believe #164 is already fixed with #209.

**Comment by jank324** [comment]
#227 will have to be fixed before release as well.

**Comment by jank324** [comment]
#241 also needs fixing before this release!

**Comment by jank324** [comment]
#246 also goes on the list.

**Comment by cr-xu** [comment]
I really think #208 should be merged asap as it is the most _breaking change_, otherwise a lot of the PRs still need to adapt to it afterwards.

**Comment by roussel-ryan** [comment]
Need to address the following before 0.7.0, sorry I don't know how to do the pretty formatting
[ ] #292 
[ ] #291

**Comment by jank324** [comment]
I would also like to sort the following out before release:
 - [ ] #267
